libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1580,6 +1580,17 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			m_context << Instruction::DUP1 << u256(32) << Instruction::ADD;
 			utils().storeStringData(contract.name());
 		}
+		else if (member == "interfaceId")
+		{
+			TypePointer arg = dynamic_cast<MagicType const&>(*_memberAccess.expression().annotation().type).typeArgument();
+			ContractDefinition const& contract = dynamic_cast<ContractType const&>(*arg).contractDefinition();
+			u256 result{0};
+			for (auto const& function: contract.interfaceFunctionList(false))
+				result = result ^ u256(function.first.asBytes());
```

Comment:
Can you use `uint64_t` for `result` and then replace this line by
```suggestion
				result ^= fromBigEndian<uint64_t>(function.first.ref());
```